crates/bevy_ecs/src/component.rs

```diff
@@ -14,26 +14,103 @@ use std::{
     mem::needs_drop,
 };
 
-/// A component is data associated with an [`Entity`](crate::entity::Entity). Each entity can have
-/// multiple different types of components, but only one of them per type.
+/// A data type that can be used to store data for an [entity].
 ///
-/// Any type that is `Send + Sync + 'static` can implement `Component` using `#[derive(Component)]`.
 ///
-/// In order to use foreign types as components, wrap them using a newtype pattern.
+/// `Component` is a [derivable trait]: this means that a data type can implement it by simply prepending a `#[derive(Component)]` attribute to it.
```

Comment:
Is the "simply" here necessary? Instead of prepending, could "apply" be said instead?

> `Component` is a [derivable trait]: this means that a data type can implement it by applying a `#[derive(Component)]` attribute to it.